Packs/Rapid7_Nexpose/Integrations/Rapid7_Nexpose/Rapid7_Nexpose_test.py

```diff
@@ -27,10 +27,35 @@ def load_test_data(folder: str, file_name: str) -> dict:
     Returns:
         dict: Dictionary data loaded from the json file.
     """
-    with open(Path("test_data") / folder / f"{file_name}.json", "r") as f:
+    with open(Path("test_data") / folder / f"{file_name}.json") as f:
         return json.load(f)
 
 
+def test_connection_errors_recovers(mocker, mock_client):
+    """
+    Given:
+     - Connection Error, ReadTimeout error and a success response
+
+    When:
+     - running the _http_request method
+
+    Then:
+     - Ensure that success message is printed and recovery for http request happens.
+    """
+    mocker.patch.object(demisto, "error")
+    mocker.patch("Rapid7_Nexpose.time.sleep")
+    mocker.patch.object(
+        BaseClient,
+        "_http_request",
+        side_effect=[
+            DemistoException(message="error", exception=requests.ConnectionError("error")),
```

Comment:
Add to the tests 
` DemistoException(message="error", exception=ValueError("error"))`
to ensure that an error will raised in other cases

Reply:
will add :)

Reply:
added